classes/class-gf-securesubmit.php

```diff
@@ -1668,4 +1656,575 @@ public function get_validation_result($validationResult, $authorizationResult)
+
+                                        $this->log_debug(__METHOD__ . '(): First Charge Approved');
+
+                                    }
+
+                                } // if
+
+
+                                // If a setup fee is required, add an invoice item.
+                                if ($single_payment_amount) {
+
+                                    $this->log_debug(__METHOD__ . '(): Processing one time setup fee');
+                                    $creditService = new HpsFluentCreditService($this->getHpsServicesConfig($this->getSecretApiKey($feed)));
+                                    /** @var HpsAuthorization $response */
+                                    /** @noinspection PhpParamsInspection */
+                                    $response = $creditService
+                                        ->recurring($single_payment_amount*100)
+                                        //->recurring(1000)
+                                        ->withPaymentMethodKey($payPlanPaymentMethod->paymentMethodKey)
+                                        ->withSchedule($planSchedule->scheduleKey)
+                                        ->execute();
+
+                                    if (!($response->transactionId > 0 && null !== $response->authorizationCode)) {
+
+                                        $this->log_debug(__METHOD__ . '(): Setup Fee Failed!! ');
+                                        $subscribResult = $this->authorization_error($userError);
+
+                                    } // if
+                                    else {
+
+                                        $this->log_debug(__METHOD__ . '(): Setup Fee Approved');
+
+                                    }
+
+                                } // if
+
+                                if (!isset($subscribResult) && null === rgar($subscribResult, 'error_message')) {
+
+                                    $subscribResult = array(
+                                        'is_success' => true,
+                                        'subscription_id' => $plan->paymentMethodKey,
+                                        'customer_id' => $customer->customerKey,
+                                        'amount' => $payment_amount,
+                                    ); // array
+
+                                } // if
+
+                            } // if
+                        }
+                    }
+                }
+
+            } catch (\Exception $e) {
+
+                // Return authorization error.
+                $subscribResult = $this->authorization_error($userError . $e->getMessage());
+
+            }
+        }
+
+        if (!isset($subscribResult)) {
+
+            $this->log_debug(__METHOD__ . '(): Unknown error ');
+            $subscribResult = $this->authorization_error($userError);
+
+        } // if
+        // Return subscription data.
+        return $subscribResult;
+
+    }
```

Comment:
All style guides and best practices I've seen prefer early returns over deeply nested code to help reduce complexity and developer cognitive load, but I would be interested in seeing any material you have that supports not returning early.

As a note, just because other implementations follow this structure, doesn't mean our implementation has to do so. 😃